homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -0,0 +1,118 @@
+"""Config flow for Minecraft Server integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+
+from . import MinecraftServer
+from .const import (  # pylint:disable=unused-import
+    CONF_UPDATE_INTERVAL,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_UPDATE_INTERVAL_SECONDS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MinecraftServerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Minecraft Server."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize Minecraft Server config flow."""
+        pass
+
+    async def async_step_user(self, user_input):
+        """Handle the initial step."""
+        if user_input is None:
+            # No configuration data available yet, show default configuration form.
+            return await self._show_config_form()
+
+        name_exists = False
+
+        # Abort in case the host name was already configured before.
+        for config_entry in self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
To check if an entry is already configured, I recommand the `config_entry.unique_id` as here : 
https://github.com/home-assistant/home-assistant/blob/3b14d9f3754ae3e31b1a3698a06b58b7498502d2/homeassistant/components/elgato/config_flow.py#L38-L40

In that way, I don't think the unique_id should contain the name because otherwise we could create 2 entries for the same host.
Do we really care if the name is different ? Usually, people will have only one server, if multiple, they would be carrefull.

So no need of "host_exists" and "name_exists", will be replaced by "already_configured", change to message in consequence.

Reply:
I'll have a look at it and come back to you. In general you're right, host name would be enough. Currently I create the sensor entities with the name given in the config flow.

Reply:
Done.